File: fabric/src/main/java/com/terrasect/fabric/TerrasectFabric.java
package com.terrasect.fabric;

import com.terrasect.common.Terrasect;
import net.fabricmc.api.ModInitializer;

public class TerrasectFabric implements ModInitializer {
    @Override
    public void onInitialize() {
        Terrasect.init();
    }
}

File: fabric/src/main/kotlin/terrasect/TerrasectFabric.kt
package terrasect

import net.fabricmc.api.ModInitializer
import net.fabricmc.loader.api.FabricLoader
import org.slf4j.LoggerFactory

object TerrasectFabric : ModInitializer {
  private val log = LoggerFactory.getLogger(Constants.MOD_ID)

  override fun onInitialize() {
    log.info("Hello from ${Constants.MOD_NAME} on Fabric!")

    Terrasect.init(FabricLoader.getInstance().configDir)
  }
}
